src/integration/Privatisation.spec.ts

```diff
@@ -0,0 +1,514 @@
+        password: newPassword,
+        enablePassword: true,
+      }
+
+      // Act
+      const actual = await request(app)
+        .post(`/${MOCK_REPO_NAME_TWO}/repo-password`)
+        .send(mockPrivatisationRequest)
+
+      // Assert
+      expect(actual.statusCode).toEqual(200)
+      const actualDeployment = await Deployment.findOne({
+        where: {
+          id: MOCK_DEPLOYMENT_DBENTRY_TWO.id,
+        },
+      })
+      if (!actualDeployment) fail()
+      const { encryptionIv, encryptedPassword } = actualDeployment
+      expect(encryptionIv).not.toEqual(MOCK_DEPLOYMENT_DBENTRY_TWO.encryptionIv)
+      expect(encryptionIv).not.toEqual(null)
+      expect(encryptedPassword).not.toEqual(
+        MOCK_DEPLOYMENT_DBENTRY_TWO.encryptedPassword
+      )
+      expect(encryptedPassword).not.toEqual(null)
+      expect(mockGenericAxios.patch).not.toHaveBeenCalled()
+    })
+
+    it("should successfully remove a password for private amplify sites", async () => {
+      // Arrange
+      const app = generateRouterForUserWithSite(
+        subrouter,
+        MOCK_USER_SESSION_DATA_TWO,
+        MOCK_REPO_NAME_TWO
+      )
+      const mockPrivatisationRequest = {
+        password: "",
+        enablePassword: false,
+      }
+
+      // Act
+      const actual = await request(app)
+        .post(`/${MOCK_REPO_NAME_TWO}/repo-password`)
+        .send(mockPrivatisationRequest)
+
+      // Assert
+      expect(actual.statusCode).toEqual(200)
+      const actualDeployment = await Deployment.findOne({
+        where: {
+          id: MOCK_DEPLOYMENT_DBENTRY_TWO.id,
+        },
+      })
+      if (!actualDeployment) fail()
+      const { encryptionIv, encryptedPassword } = actualDeployment
+      expect(encryptionIv).toEqual(null)
+      expect(encryptedPassword).toEqual(null)
+      expect(mockGenericAxios.patch).toHaveBeenCalled()
+    })
+
+    it("should do nothing when removing a password for already public amplify sites", async () => {
+      // Arrange
+      const app = generateRouterForUserWithSite(
+        subrouter,
+        MOCK_USER_SESSION_DATA_ONE,
+        MOCK_REPO_NAME_ONE
+      )
+      const mockPrivatisationRequest = {
+        password: "",
+        enablePassword: false,
+      }
+
+      // Act
+      const actual = await request(app)
+        .post(`/${MOCK_REPO_NAME_ONE}/repo-password`)
+        .send(mockPrivatisationRequest)
+
+      // Assert
+      expect(actual.statusCode).toEqual(200)
+      const actualDeployment = await Deployment.findOne({
+        where: {
+          id: MOCK_DEPLOYMENT_DBENTRY_ONE.id,
+        },
+      })
+      if (!actualDeployment) fail()
+      const { encryptionIv, encryptedPassword } = actualDeployment
+      expect(encryptionIv).toEqual(null)
+      expect(encryptedPassword).toEqual(null)
+      expect(mockGenericAxios.patch).not.toHaveBeenCalled()
+    })
+
+    it("should successfully set a password for public amplify sites", async () => {
```

Comment:
will it cause the sites to become private? if so, how are we tsting this?

Reply:
I'm checking this via the `mockAxios.patch` call, which only occurs for this route when the github repo is changed (to make private or public)